src/main/java/tigase/server/rtbl/RTBLRepository.java

```diff
@@ -0,0 +1,459 @@
+/*
+ * Tigase XMPP Server - The instant messaging server
+ * Copyright (C) 2004 Tigase, Inc. (office@tigase.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. Look for COPYING file in the top folder.
+ * If not, see http://www.gnu.org/licenses/.
+ */
+package tigase.server.rtbl;
+
+import tigase.db.TigaseDBException;
+import tigase.db.UserRepository;
+import tigase.eventbus.EventBus;
+import tigase.eventbus.EventBusEvent;
+import tigase.eventbus.HandleEvent;
+import tigase.kernel.beans.Bean;
+import tigase.kernel.beans.Initializable;
+import tigase.kernel.beans.Inject;
+import tigase.kernel.beans.UnregisterAware;
+import tigase.kernel.beans.config.ConfigField;
+import tigase.kernel.core.Kernel;
+import tigase.stats.StatisticsContainerIfc;
+import tigase.stats.StatisticsList;
+import tigase.xmpp.jid.BareJID;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+@Bean(name = "rtblRepository", parent = Kernel.class, active = true, exportable = true)
+public class RTBLRepository implements Initializable, UnregisterAware, StatisticsContainerIfc {
+	
+	private static final Logger logger = Logger.getLogger(RTBLRepository.class.getCanonicalName());
+
+	@Inject
+	private UserRepository userRepository;
+	@Inject(bean = "eventBus")
+	private EventBus eventBus;
+
+	@ConfigField(desc = "Reload interval")
+	private long reloadInterval = 10 * 60 * 1000;
```

Comment:
Maybe for readability: `TimeUnit.MINUTES.toMillis(10);`?

Reply:
We can always improve and don't have to follow to the point what was already in other places...